examples/zephyr/main.c

```diff
@@ -0,0 +1,130 @@
+#include <string.h>
+#include <zephyr/kernel.h>
+#include "note-c/note.h"
+#include "note_c_hooks.h"
+#include "NotecardEnvVarManager.h"
+
+/*
+ * Uncomment this line and replace com.your-company:your-product-name with your
+ * ProductUID.
+ */
+/* #define PRODUCT_UID "com.your-company:your-product-name" */
+
+#ifndef PRODUCT_UID
+#define PRODUCT_UID ""
+#pragma message "PRODUCT_UID is not defined in this example. Please ensure your Notecard has a product identifier set before running this example or define it in code here. More details at https://bit.ly/product-uid"
+#endif
+
+/* 5 second timeout for retrying the hub.set request. */
+#define HUB_SET_RETRY_SECONDS 5
+/* Fetch every 20 seconds. */
+#define FETCH_INTERVAL_SECONDS 20
+
+/* A struct to cache the values of environment variables. */
+typedef struct {
+    char valueA[16];
+    char valueB[16];
+    char valueC[16];
+} EnvVarCache;
+
+NotecardEnvVarManager *envVarManager = NULL;
+EnvVarCache envVarCache;
+
+/* These are the environment variables we'll be fetching from the Notecard. */
+const char *envVars[] = {
+    "variable_a",
+    "variable_b",
+    "variable_c"
+};
+static const size_t numEnvVars = sizeof(envVars) / sizeof(envVars[0]);
+
+struct k_work envUpdateWorkItem;
+struct k_timer envUpdateTimer;
+
+/*
+ * Callback that will be executed when the environment variable update timer
+ * expires.
+ */
+static void envUpdateTimerCb(struct k_timer *timer)
+{
+    k_work_submit(&envUpdateWorkItem);
+}
+
+/*
+ * Callback that will be executed by the system workqueue when it's time to
+ * check for environment variable updates.
+ */
+static void envUpdateWorkCb(struct k_work *item)
+{
+    NotecardEnvVarManager_fetch(envVarManager, envVars, numEnvVars);
+}
+
+void envVarManagerCb(const char *var, const char *val, void *userCtx)
+{
+    EnvVarCache *cache = (EnvVarCache *)userCtx;
+
+    /* Cache the values for each variable. */
+    if (strcmp(var, "variable_a") == 0) {
+        strncpy(cache->valueA, val, sizeof(cache->valueA));
+    } else if (strcmp(var, "variable_b") == 0) {
+        strncpy(cache->valueB, val, sizeof(cache->valueB));
+    } else if (strcmp(var, "variable_c") == 0) {
+        strncpy(cache->valueC, val, sizeof(cache->valueC));
+    }
+
+    printk("variable_a has value %s\n", envVarCache.valueA);
+    printk("variable_b has value %s\n", envVarCache.valueB);
+    printk("variable_c has value %s\n", envVarCache.valueC);
```

Comment:
We can make the log output easier to digest without requiring source code context.

```suggestion
    printk("\nCallback received variable (%s) with value (%s) and context (%p).\n", var, val, userCtx);

    /* Cache the values for each variable. */
    if (strcmp(var, "variable_a") == 0) {
        strncpy(cache->valueA, val, sizeof(cache->valueA));
    } else if (strcmp(var, "variable_b") == 0) {
        strncpy(cache->valueB, val, sizeof(cache->valueB));
    } else if (strcmp(var, "variable_c") == 0) {
        strncpy(cache->valueC, val, sizeof(cache->valueC));
    }

    /* Print cached values */
    printk("Cached values:\n");
    printk("- variable_a has value %s\n", envVarCache.valueA);
    printk("- variable_b has value %s\n", envVarCache.valueB);
    printk("- variable_c has value %s\n", envVarCache.valueC);
```

I looked at the output before I read the code, and the repetition was not immediately obvious...

```none
{"req":"env.get","names":["variable_a","variable_b","variable_c"]}
{"body":{}}
{"req":"env.get","names":["variable_a","variable_b","variable_c"]}
{"body":{"variable_a":"Good","variable_b":"job","variable_c":"Hayden!"}}
variable_a has value Good
variable_b has value 
variable_c has value 
variable_a has value Good
variable_b has value job
variable_c has value 
variable_a has value Good
variable_b has value job
variable_c has value Hayden!
{"req":"env.get","names":["variable_a","variable_b","variable_c"]}
{"body":{"variable_a":"Good","variable_b":"job","variable_c":"Hayden!"}}
variable_a has value Good
variable_b has value job
variable_c has value Hayden!
variable_a has value Good
variable_b has value job
variable_c has value Hayden!
variable_a has value Good
variable_b has value job
variable_c has value Hayden!
```